odoo/addons/base/tests/test_tests_tags.py

```diff
@@ -0,0 +1,234 @@
+                pass
+            
+        with self.assertRaises(TagsError):
+            @tagged('js+slow')
+            class FakeClass:
+                pass
+
+@tagged('nodatabase')
+class TestSelector(unittest.TestCase):
+
+    def test_selector_parser(self):
+        """Test the parser part of the TagsTestSelector class"""
+
+        selector = TagsTestSelector('+slow')
+        self.assertEqual({'slow', }, selector.include)
+        self.assertEqual(set(), selector.exclude)
+
+        selector = TagsTestSelector('+slow,nightly')
+        self.assertEqual({'slow', 'nightly'}, selector.include)
+        self.assertEqual(set(), selector.exclude)
+
+        selector = TagsTestSelector('+slow,-standard')
+        self.assertEqual({'slow', }, selector.include)
+        self.assertEqual({'standard', }, selector.exclude)
+
+        # same with space after the comma
+        selector = TagsTestSelector('+slow, -standard')
+        self.assertEqual({'slow', }, selector.include)
+        self.assertEqual({'standard', }, selector.exclude)
+
+        # same with space befaore and after the comma
+        selector = TagsTestSelector('+slow , -standard')
+        self.assertEqual({'slow', }, selector.include)
+        self.assertEqual({'standard', }, selector.exclude)
+
+        selector = TagsTestSelector('+slow ,-standard,+js')
+        self.assertEqual({'slow', 'js', }, selector.include)
+        self.assertEqual({'standard', }, selector.exclude)
+
+        selector = TagsTestSelector('slow, ')
+        self.assertEqual({'slow', }, selector.include)
+        self.assertEqual(set(), selector.exclude)
+
+        selector = TagsTestSelector('+slow,-standard, slow,-standard ')
+        self.assertEqual({'slow', }, selector.include)
+        self.assertEqual({'standard', }, selector.exclude)
+
+        selector = TagsTestSelector('')
+        self.assertEqual({'standard', }, selector.include)
+        self.assertEqual(set(), selector.exclude)
+
+@tagged('nodatabase')
+class TestSelectorSelection(unittest.TestCase):
+
+    def test_selector_selection(self):
+        """Test check_tags use cases"""
+        class Test_A():
+            pass
+
+        @tagged('stock')
+        class Test_B():
+            pass
+
+        @tagged('stock', 'slow')
+        class Test_C():
+            pass
+
+        @tagged('standard', 'slow')
+        class Test_D():
+            pass
+
+        no_tags_obj = Test_A()
+        stock_tag_obj = Test_B()
+        multiple_tags_obj = Test_C()
+        multiple_tags_standard_obj = Test_D()
+
+        # if 'standard' in not explicitly removed, tests without tags are
+        # considered tagged standards and they are run by default if
+        # not explicitly deselected with '-standards' or if 'standards' is not
+        # selectected along with another test tag
+
+        # same as no "--test-tags" parameters:
+        selector = TagsTestSelector('')
+        self.assertTrue(selector(no_tags_obj))
```

Comment:
As the selector is used as a function, I would call it `select`. What do you think?